### PR TITLE
Correctly order lessons in workshop pre-survey results graph

### PR DESCRIPTION
**Pull request by aoby**
[Task](https://trello.com/c/GbK8yrpc/163-correctly-alphabetize-graph-in-workshop-pre-survey)

### Before
Note, within unit 4, lesson 10 comes before lesson 2 since the ordering is purely alphabetical.
image

### After
Now, the lessons are correctly ordered: 1, 2, 10
image

I also added unit tests, both for this new sorting feature and basic functionality for this control.

### CODE REVIEW

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          school: "A school",
+          user_id: userIndex,
+          attended: true,
+          pre_workshop_survey: survey
+        };
+      };
+
+      fakeEnrollments = [
+        generateFakeEnrollment({
+          unit: "Unit 4 - the fouth unit",
```
Comment (clareconstantine) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: tiny detail: missing the 'r' in 'fourth'

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+          unitLessonShortName: "U1 L2"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 1 - the first lesson",
+          questionsAndTopics: "",
+          unitLessonShortName: "U1 L1"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 4 - the fouth unit",
```
Comment (clareconstantine) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: also missing an 'r' here

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+          lesson: "Lesson 3 - the third lesson",
+          questionsAndTopics: "more questions...",
+          unitLessonShortName: "U4 L3"
+        }),
+        // one no response
```
Comment (clareconstantine) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: should that be `on no response`?

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+    it("Displays survey responses", () => {
+      const ROW_FULL_RESPONSE = 0;
+      const responseCellText = tableRows.at(ROW_FULL_RESPONSE).find("td").map(td => td.text());
+      expect(responseCellText).to.eql([
+        `${ROW_FULL_RESPONSE + 1}`,
+        fakeEnrollments[ROW_FULL_RESPONSE].first_name,
+        fakeEnrollments[ROW_FULL_RESPONSE].last_name,
+        fakeEnrollments[ROW_FULL_RESPONSE].email,
+        "Unit 4 - the fouth unit",
```
Comment (clareconstantine) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: is 'fouth' a reference i don't understand? sorry if these were all intentional and i'm just missing the joke!

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx (hunk 3 above)
Comment (aoby) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: No, I meant one enrollment with no response. How about that? `"One enrollment with no response"`? Or maybe it's self-explanatory enough without the comment?

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx (hunk 4 above)
Comment (aoby) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: Copy/paste error. Not intentional. Though now I want to come up with something... ;)

Comment (aoby) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx]: Or rather, a successful copy/paste from the original typo

**Pull request merged**